src/views/client/index.tsx

```diff
@@ -17,15 +18,31 @@ import Game from "./game";
 import WaitingRoom from "./waitingRoom";
 
 interface Props {
-  deviceSize: {
-    width: number;
-    height: number;
-  };
+  device: any;
   isAdmin: boolean;
+  adminRoomId?: string;
+  autoconnect?: boolean;
+  onCreateRoom: any;
+  adminPosition: number;
 }
 
-const Client: FC<Props> = ({ deviceSize, isAdmin }) => {
+const Client: FC<Props> = ({
+  device,
+  isAdmin,
+  adminRoomId,
+  autoconnect,
+  onCreateRoom,
+  adminPosition
+}) => {
   const dispatch = useDispatch();
+  const roomId = useSelector(selectors.room.selectId);
+
+  const handleOnCreateRoom = useCallback(
+    (adminRoomId?: string) => {
+      onCreateRoom(adminRoomId);
```

Comment:
Du fait que `onCreateRoom` soit optionnel, dans le cas ou il ne sera pas passé en props (en dehors de l'admin donc) on aura une erreur ici `TypeError: onCreateRoom is not a function`, d'où l'importance du typage 🙂